src/compiler/types.ts

```diff
@@ -3371,8 +3371,8 @@ namespace ts {
     export interface Symbol {
         flags: SymbolFlags;                     // Symbol flags
         escapedName: __String;                  // Name of symbol
-        declarations: Declaration[];           // Declarations associated with this symbol
-        valueDeclaration: Declaration;         // First value declaration of the symbol
+        declarations: Declaration[];            // Declarations associated with this symbol
+        valueDeclaration: Declaration;          // First value declaration of the symbol

```

Comment:
I still don't think this is the correct thing to do @andy-ms 

> Yes, the change was to remove a large number of assertions even if it meant having a less accurate type declaration.

If that property was internal, I wouldn't have a problem with that. But this makes the API really dangerous to use. Using `--strictNullChecks` in the compiler's code base should actually increase type safety for API users instead of decreasing it.

With the same reasoning you could make `Node.parent` non-optional to safe a lot of assertions, because it will be defined *almost* all the time.

Reply:
It's possible that we could change the output `.d.ts` files in a postprocessing step. CC @weswigham